src/utils/polygon.cpp

```diff
@@ -100,6 +101,28 @@ Polygons Polygons::approxConvexHull(int extra_outset)
     return convex_hull.unionPolygons().offset(-overshoot + extra_outset, ClipperLib::jtRound);
 }
 
+void Polygons::makeConvex()
+{
+    for (PolygonRef poly : *this)
+    {
+        Point a = poly.back();
+        for (int i = 0; i < poly.size();)
+        {
+            Point b = poly[i];
+            Point c = poly[(i + 1) % poly.size()];
+            if (LinearAlg2D::pointIsLeftOfLine(b, a, c) >= 0)
+            {
+                poly.remove(i);
```

Comment:
We might want to copy the data to a separate polygon and then `std::swap` for better theoretical performance and simplification of this code.

Reply:
I've implemented this now.